Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -7,401 +7,882 @@
  */
 
 #include <AzCore/Console/IConsole.h>
+#include <AzCore/Date/DateFormat.h>
+#include <AzCore/Debug/StackTracer.h>
+#include <AzCore/IO/GenericStreams.h>
 #include <AzCore/Math/Crc.h>
+#include <AzCore/Memory/AllocationRecords.h>
 #include <AzCore/Memory/AllocatorManager.h>
 #include <AzCore/Memory/Memory.h>
 #include <AzCore/Memory/IAllocator.h>
-
 #include <AzCore/Memory/OSAllocator.h>
-#include <AzCore/Memory/AllocationRecords.h>
+
+#include <AzCore/Platform.h>
 
 #include <AzCore/std/parallel/lock.h>
-#include <AzCore/std/smart_ptr/make_shared.h>
 #include <AzCore/std/containers/array.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzCore/Utils/Utils.h>
 
 namespace AZ
 {
+    static void sys_DumpAllocators([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
+    {
+        AllocatorManager::Instance().DumpAllocators();
+    }
+    AZ_CONSOLEFREEFUNC(sys_DumpAllocators, AZ::ConsoleFunctorFlags::Null, "Print memory allocator statistics.");
 
-static void sys_DumpAllocators([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
-{
-    AllocatorManager::Instance().DumpAllocators();
-}
-AZ_CONSOLEFREEFUNC(sys_DumpAllocators, AZ::ConsoleFunctorFlags::Null, "Print memory allocator statistics.");
+    // Provides a range of allocations to dump. The min value is inclusive and the max value is exclusive
+    // Theerefore the range is [min, max)
+    struct AllocationDumpRange
+    {
+        size_t m_min = 0;
+        size_t m_max = AZStd::numeric_limits<size_t>::max();
+    };
 
-static EnvironmentVariable<AllocatorManager>& GetAllocatorManagerEnvVar()
-{
-    static EnvironmentVariable<AllocatorManager> s_allocManager;
-    return s_allocManager;
-}
+    // Add console command for outputting the Allocation records for all or a specified set of allocators with the provided names
+    static void DumpAllocationsForAllocatorHelper(
+        AZStd::span<AZStd::string_view const> allocatorNameArguments,
+        AZ::IO::GenericStream& printStream,
+        const AllocationDumpRange& allocationDumpRange = {})
+    {
+        auto& allocatorManager = AZ::AllocatorManager::Instance();
+        const int numAllocators = allocatorManager.GetNumAllocators();
+        // Garbage collect the allocations before dumping them
+        allocatorManager.GarbageCollect();
```

Comment:
Good call. Might be a useful option to print out garbage info in a future PR (if we collect that info like how min/max/avg garbage memory size).